app/bundles/CoreBundle/Translations/en_US/validators.ini

```diff
@@ -1,5 +1,6 @@
 mautic.core.ab_test.winner_criteria.not_blank="Select a winning criteria."
 mautic.core.email.required="A valid email is required."
+mautic.core.segment.circular_dependency_exists="Update cannot be completed as there are one or more circular dependencies in this segment. Please remove the dependency for the segment to function."
```

Comment:
I would suggest changing this to something along the lines of: `Update cannot be completed as the current filters are conflicting with the filters of another segment. Make sure your segment filters are not creating impossible or contradictory conditions (e.g. Segment A must be a member of Segment B while Segment B must be a member of Segment A).`